backend/backend/container.py

```diff
@@ -15,128 +24,175 @@ class Status(Enum):
             return Status.NOTFOUND
         else:
             if sts == "running":
                 return Status.RUNNING
-            elif sts == "exited":
-                return Status.EXITED
             elif sts == "restarting":
                 return Status.RESTARTING
             elif sts == "paused":
                 return Status.PAUSED
-            else:
-                return Status.UNDEFINED
-
-    def create_shared_folder(self, id: int, user: str):
-        """Creates a directory for the docker container with the specified id,
-        exposes the specified SSH port, and has SSH login credentials user/password.
-        Inside it creates a script for initialization that uses the specified username
-        for the home directory.
-
-        :param id: ID (name) of container
-        :param port: SSH port that is exposed
-        :param user: Username for SSH
-        :param password: Password for SSH
+            elif sts == "exited":
+                return Status.EXITED
+        return Status.UNDEFINED
+
+    def format_config(self, container_id: int, user: str, password: str,
+                      hostname='Dell-T140', user_id='1000', group_id='1000',
+                      timezone='Europe/London', sudo_access='true',
+                      image='ghcr.io/linuxserver/openssh-server', port=None) -> dict:
+        """Formats the given parameters as a dictionary that fits docker-py.
+        Creates the volumes for the config and home dirs of the container
+
+        :param container_id: Unique ID for container
+        :param user: Username for container
+        :param password: Password for container
+        :param volumes: Volumes on host to mount to the container.
+        :param hostname: Hostname for container, defaults to 'Dell-T140'
+        :param user_id: UID for container user, defaults to '1000'
+        :param group_id: GID for container user, defaults to '1000'
+        :param timezone: Timezone for container, defaults to 'Europe/London'
+        :param sudo_access: Sudo access for container, defaults to 'true'
+        :param image: Image for container, defaults to 'ghcr.io/linuxserver/openssh-server'
+        :param port: Exposed port for container, defaults to None
+        :return: Dictionary that can be easily used for docker-py
         """
-        # Save current working directory, must be able to restore later
-        current_dir = os.getcwd()
 
-        # Makes a shared folder named with the id
-        os.mkdir(str(id))
+        # Format the container id to ID_PREFIX + id
+        _container_id = Containers.ID_PREFIX + str(container_id)
+        _config_path = _container_id + "config"
+        _home_path = _container_id + "home"
+
+        # Format the config dict of this container
+        config = {'Image': image, 'ID': _container_id, 'Environment': self.format_environment(
+            user, password, user_id, group_id, timezone, sudo_access),
+            'Port': {'2222/tcp': port},
```

Comment:
Is it that port 2222 is bound inside the container and the `port` variable is a host port where everything is redirected to 2222 inside the container? If it is, could we maybe change it to port 22 since the attackers will be attacking the frontend on port 22 and would expect the SSH server to also be listening on port 22

Reply:
Yes, it exposes a "random" port (assigned by the OS) externally. I have not been successful in binding port 22 internally to the container since the container refuses to start with error regarding the log files.
![image](https://user-images.githubusercontent.com/10566507/112841580-f258b900-90a0-11eb-9d66-3c2505f5b9b0.png)
I do not know how big of a problem it will be to have this running on 2222, but it may show up if they inspect netstat.
![image](https://user-images.githubusercontent.com/10566507/112842113-81fe6780-90a1-11eb-9118-e8ec2e04b1e5.png)